nodebuilder/core/module.go

```diff
@@ -0,0 +1,49 @@
+package core
+
+import (
+	"context"
+
+	"go.uber.org/fx"
+
+	"github.com/celestiaorg/celestia-node/core"
+	"github.com/celestiaorg/celestia-node/header"
+	headercore "github.com/celestiaorg/celestia-node/header/core"
+	"github.com/celestiaorg/celestia-node/libs/fxutil"
+	"github.com/celestiaorg/celestia-node/nodebuilder/node"
+)
+
+// Module collects all the components and services related to managing the relationship with the Core node.
+func Module(tp node.Type, cfg *Config, options ...fx.Option) fx.Option {
+	switch tp {
+	case node.Light, node.Full:
+		return fx.Module("core", fx.Supply(*cfg), fx.Options(options...))
```

Comment:
Probably typo missing `fx.Invoke(cfg.ValidateBasic)` here

Reply:
Yup, good catch

Reply:
True, good catch! we should still invoke basic config validation on full/light even if we don't need the core options since we actually read from core config for state module, cc @distractedm1nd . I'll implement the fix.